packages/libp2p/src/libp2p.ts

```diff
@@ -98,8 +98,14 @@ export class Libp2pNode<T extends ServiceMap = Record<string, unknown>> extends
 
     components.events.addEventListener('peer:update', evt => {
       // if there was no peer previously in the peer store this is a new peer
-      if (evt.detail.previous == null) {
-        this.safeDispatchEvent('peer:discovery', { detail: evt.detail.peer })
+      if (evt.detail.previous === null) {
```

Comment:
```suggestion
      if (evt.detail.previous == null) {
```

Reply:
the emphasis on the double equal is because it's is it possible this value could also be `undefined` ?

Reply:
Yes, in this case == equals null or undefined, === equals only null.